src/widget-core/decorators/customElement.ts

```diff
@@ -37,22 +37,22 @@ export interface CustomElementConfig<P extends object = { [index: string]: any }
  * This Decorator is provided properties that define the behavior of a custom element, and
  * registers that custom element.
  */
-export function customElement<P extends object = { [index: string]: any }>({
-	tag,
-	properties = [],
-	attributes = [],
-	events = [],
-	childType = CustomElementChildType.DOJO,
-	registryFactory = () => new Registry()
-}: CustomElementConfig<P>) {
+export function customElement<P extends object = { [index: string]: any }>(config: CustomElementConfig<P>) {
+	const defaults = {
+		properties: [],
+		attributes: [],
+		events: [],
+		childType: CustomElementChildType.DOJO,
+		registryFactory: () => new Registry()
+	};
+	const { tag: tagName, ...configRest } = config;
+	const updatedConfig = { ...(tagName ? { tagName } : {}), ...configRest };
```

Comment:
I find this really hard to follow what exactly is going on here... Perhaps it could be more verbose but end up clearer?

Reply:
@agubler I made some subtle changes here and added a comment. I mostly just removed the ternary (your comment below about undefined means this should destructure fine as an empty object if tag name isn't specified).  Anyways, hopefully this should be a little clearer now.